app/models/account/data_enricher.rb

```diff
@@ -8,49 +8,61 @@ def initialize(account)
   end
 
   def run
-    enrich_transactions
-  end
+    total_unenriched = account.entries.account_transactions
+      .where("enriched_at IS NULL OR merchant_id IS NULL OR category_id IS NULL")
```

Comment:
Both `merchant_id` and `category_id` are stored on the `Account::Transaction` record, while `enriched_at` is stored on `Account::Entry` (delegated type).

Given we're primarily enriching `Account::Transaction` records (and not other entry types like `Account::Valuation` and `Account::Trade`), I think it probably makes sense to _move_ the `enriched_at` field down to the `Account::Transaction` and only deal with those types of records in this process.

Reply:
Actually, second guessing the idea of moving `enriched_at` to `Account::Transaction`.  This enrichment is also modifying `enriched_name` on the `Account::Entry` model, so it probably makes sense to keep this as-is.

Will still need to update this query to read the filters off the correct DB records though.